sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2747,6 +2747,17 @@ class Dataset[T] private[sql](
     }
   }
 
+  /**
+   * Collect a Dataset as ArrowPayload byte arrays and serve to PySpark.
+   */
+  private[sql] def collectAsArrowToPython(): Int = {
+    val payloadRdd = toArrowPayloadBytes()
+    val payloadByteArrays = payloadRdd.collect()
```

Comment:
@BryanCutler Btw, it is not for performance gain I think. `toLocalIteratorAndServe` can avoid collect all data at once into the driver. So it may be good for the memory usage on the driver side.